src/vecs/client.py

```diff
@@ -96,12 +96,33 @@ def get_collection(self, name: str) -> Collection:
             CollectionNotFound: If no collection with the given name exists.
         """
         from vecs.collection import Collection
-
-        collections = Collection._list_collections(self)
-        for collection in collections:
-            if collection.name == name:
-                return collection
-        raise CollectionNotFound("No collection found with requested name")
+        query = text(
+        f"""
+        select
+            relname as table_name,
+            atttypmod as embedding_dim
+        from
+            pg_class pc
+            join pg_attribute pa
+                on pc.oid = pa.attrelid
+        where
+            pc.relnamespace = 'vecs'::regnamespace
+            and pc.relkind = 'r'
+            and pa.attname = 'vec'
+            and not pc.relname ^@ '_'
+            and pc.relname = '{name}'
+        """
+        )
```

Comment:
this should work but could you pls update it to use a [parameterized query](https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.text) to protect against [SQL injection](https://en.wikipedia.org/wiki/SQL_injection)?

Reply:
should be something like 
```python
...
text("""
...
and pc.relname = :name
""").bindparams(name=name)
```